crypto/src/mls/conversation/decrypt.rs

```diff
@@ -475,7 +477,12 @@ impl CentralContext {
             .await;
 
         let decrypt_message = match decrypt_message {
-            Err(Error::BufferedFutureMessage) => self.handle_future_message(id, message).await?,
+            error @ Err(Error::BufferedFutureMessage { .. }) => {
+                self.handle_future_message(id, message).await?;
+                let Err(Error::BufferedFutureMessage { message_epoch }) = error;
+                info!("Buffered future message from epoch {message_epoch}");
+                return error;
+            }
             _ => decrypt_message?,
         };
```

Comment:
Not really nice, but should fix the compile error. 

```suggestion
 let decrypt_message = match decrypt_message {
            Err(Error::BufferedFutureMessage { message_epoch }) => {
                self.handle_future_message(id, message).await?;
                info!("Buffered future message from epoch {message_epoch}");
                return decrypt_message;
            }
            _ => decrypt_message?,
        };
```